liver_imaging_analysis/models/liver_segmentation.py

```diff
@@ -680,188 +669,223 @@ def train_liver(
         Expects "2D" for slice inference, "3D" for volume inference,
         or "sliding_window" for sliding window inference.
         Default is 3D
+    automate: bool
+        flag to use automatic hyperparameter optimization.
+        Default is False
     pretrained : bool
         if true, loads pretrained checkpoint. Default is True.
     cp_path : str
-        determines the path of the checkpoint to be loaded 
-        if pretrained is true. If not defined, the potential 
+        determines the path of the checkpoint to be loaded
+        if pretrained is true. If not defined, the potential
         cp path will be loaded from config.
     epochs : int
         number of training epochs.
         If not defined, epochs will be loaded from config.
     evaluate_epochs : int
         The number of epochs to evaluate model after. Default is 1.
     batch_callback_epochs : int
-        The frequency at which per_batch_callback will be called. 
+        The frequency at which per_batch_callback will be called.
         Expects a number of epochs. Default is 100.
     save_weight : bool
         whether to save weights or not. Default is True.
     save_path : str
         the path to save weights at if save_weights is True.
-        If not defined, the potential cp path will be loaded 
+        If not defined, the potential cp path will be loaded
         from config.
     test_batch_callback : bool
         whether to call per_batch_callback during testing or not.
         Default is False
     """
-    if cp_path is None:
-        cp_path = config.save["potential_checkpoint"]
-    if epochs is None:
-        epochs = config.training["epochs"]
-    if save_path is None:
-        save_path = config.save["potential_checkpoint"]
+
     set_seed()
     model = LiverSegmentation(modality, inference)
     model.load_data()
+    model.exp_naming()
```

Comment:
Did this come from another branch as well?

Reply:
yes, I removed it